nion/swift/ExportDialog.py

```diff
@@ -241,9 +241,14 @@ def __init__(self, document_controller: DocumentController.DocumentController, d
 
         def ok_clicked() -> bool:
             dpi = 96
-            width_px = (handler.width_model.value or display_size.width) * dpi
-            height_px = (handler.height_model.value or display_size.height) * dpi
+            if display_item.display_data_shape is not None:
+                display_item_ratio = display_item.display_data_shape[1] / display_item.display_data_shape[0]
```

Comment:
This line (`display_item_ratio = ...`) makes the assumption that the display data has two dimensions, resulting in failure if you try to export a 1D line plot to SVG. The logic for the `display_size` is already available on lines 235-238.

I suggest the following:
- update the logic on lines 235-238 to default to a height of 4 for both 1D and 2D cases
- in the 2D case, use the the data aspect ratio to determine the width
- in the 2D case, limit aspect ratio to 32:1 so that an image that is 2048x1 doesn't crash SVG (this may need more thinking)
- in the 1D case, use the golden ratio as the aspect ratio, width = 4 * 1.61803, height = 4
- calculate the display_item_ratio immediately after line 238
- recalculate the height_px, width_px size where you do now lines 250-251
- this is another potential spot to limit the SVG export size, maybe no larger than 16384 pixels in either dimension?